src/contracts/bancor.js

```diff
@@ -0,0 +1,136 @@
+
+function getBancorGasPrice(web3, callback){
+		var gaslimit_contract =  new web3.eth.Contract(bancor_gaslimit_abi,bancor_gaslimit_address);
+		gaslimit_contract.methods.gasPrice().call().then((gasprice) => callback(gasprice));
+}
+
+function makeBancorPath(fromToken, toToken){
+	var path_symbols;
+	if(fromToken == "ETH"){
+		path_symbols = [fromToken,"BNT","BNT", toToken+"BNT",toToken];
+	}
+	else if(fromToken == "BNT"){
+		path_symbols = [fromToken, toToken+"BNT",toToken];
+	}
+	else{
+		path_symbols = [fromToken, fromToken+"BNT","BNT",toToken+"BNT",toToken]; 
+	}
+	var addresses = [];
+	for(var i=0;i<path_symbols.length;i++){
+		addresses.push(currencyinfo[path_symbols[i]].details[0].blockchainId);
+	}
+	return addresses;
+}
+function approve(web3,fromAddress, tokenAddress, spender, amount){
+	var token_contract =  new web3.eth.Contract(ERC20_abi,tokenAddress);
+	var data= token_contract.methods.approve(spender,amount).encodeABI()
+	var rawTransaction = {
+				"from": fromAddress,
+				"to": tokenAddress,
+				"gas": 2000000,
+				"data":data
+			};
+	return rawTransaction;	
+}
+/**
+ * 
+ * @param {*} web3 
+ * @param {*} fromAddress 
+ * @param {*} productId 
+ * @param {*} fromCurrency 
+ * @param {*} amount 
+ * @param {*} minSeconds 
+ * @param {*} gasPrice 
+ * @param {*} transactionExecutor this function will be passed raw transactions to be executed
+ */
+function buySubscriptionUsingBancor(web3,fromAddress,productId, fromCurrency,amount,minSeconds,gasPrice,transactionExecutor){
+	var path = makeBancorPath(fromCurrency,"DATA");
+	var ba_contract =  new web3.eth.Contract(ba_abi,ba_add);
+	var data;
+	var value=0;
+	if(fromCurrency == "ETH"){
+		data = ba_contract.methods.buyWithETH(productId,path,minSeconds).encodeABI();
+		value = amount;
+	}
+	else{
+		var approval = approve(web3,fromAddress,path[0],ba_add,amount);
+		console.log("Purchases with token require approval: ");
+		transactionExecutor(approval);
+		//var amount_hex = web3.utils.toHex(amount);
+		data = ba_contract.methods.buyWithERC20(productId,path,minSeconds,amount).encodeABI();
+	}
+	var rawTransaction = {
+		"from": fromAddress,
+		"to": ba_add,
+		"gas": 2000000,
+		"data":data,
+		"gasPrice":gasPrice,
+		"value": value
+	};
+	transactionExecutor(rawTransaction);
+}
+/////////////////////////////////
+
+
+module.exports = {
+    makeBancorPath, buySubscriptionUsingBancor,getBancorGasPrice
+}
+
+
+/*
```

Comment:
Need to keep this commented-out stuff here?

Reply:
I can remove if you prefer. It shows example usage, which I thought was useful